src/nimib/sources.nim

```diff
@@ -101,20 +100,30 @@ func getCodeBlock*(source: string, command: string, startPos, endPos: Pos): stri
           notIndentLines.add i
       inc i
       
-
-
     let parsedLines = collect(newSeqOfCap(codeLines.len)):
       for i in 0 .. codeLines.high:
         if i in notIndentLines:
           codeLines[i]
         else:
           codeLines[i].substr(indent)
     codeText = parsedLines.join("\n")
-    #codeText = codeLines.mapIt(it.substr(indent)).join("\n")
 
   else: # single line case, eg `nbCode: echo "Hello World"`
     let line = lines[startLine]
-    codeText = line.split(":")[1 .. ^1].join(":").strip() # split at first ":" and take the rest as code and then strip it.
+    var extractedLine = line[startPos.column .. ^1].strip()
+    if extractedLine.strip().endsWith(")"):
+      # check if the ending ")" has a matching "(", otherwise remove it.
+      var nOpen, nClose: int
+      for c in extractedLine:
+        if c == '(': nOpen += 1
+        elif c == ')': nClose += 1
+      if nOpen < nClose:
+        # remove trailing ")" so that we get matching number of "()".
+        # We trust the compiler to only give us valid code, thus we can assume all ")" are at the end as the missing "("
```

Comment:
if you're parsing code char by char, to handle edge cases you end up needing more of a parser, i'm afraid,
you end up tripping over parens inside of quotes, and then tripping over escaped quotes.
[here](https://github.com/nim-lang/Nim/blob/a78ee8ae8453aabde04fa0103e02378efe870827/lib/pure/strformat.nim#L601-L628) is what i had to do in strformat

Reply:
Ugh, you are right, nice catch 😄🙈 

That looks like quite a lot of work to find all edge cases 😅 What do you think about just counting the opening brackets before `startPos.column` until we hit a non-whitespace character that isn't an open parentheses? Or are there any other characters which are valid to put where I mark in this code?

```nim
nbCode ( echo "y" )
#            ↑ ↑
```